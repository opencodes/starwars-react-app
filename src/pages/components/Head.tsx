import React, { useState } from 'react'
import { Link } from 'react-router-dom'

type Props = {
    onSearch: (search: string) => void
}

const Head = ({ onSearch }: Props) => {
    const [search, setSearch] = useState('')
    return (
        <nav className="navbar navbar-expand-lg bg-light">
            <div className="container">
                <Link to={`/`} className="navbar-brand" >Star War</Link>
                <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>
                <div className="collapse navbar-collapse" id="navbarSupportedContent">
                    <ul className="navbar-nav me-auto mb-2 mb-lg-0">
                        <li className="nav-item">
                            <Link to={`/favourite`} >
                                Favourite
                            </Link>
                        </li>
                    </ul>
                    <form className="d-flex" role="search">
                        <input className="form-control me-2" type="search"
                            placeholder="Search" aria-label="Search"
                            data-testid="searchinput"
                            value={search}
                            onChange={(e) => setSearch(e.target.value)}
                        />
                        <button className="btn btn-outline-success" type="button"
                            data-testid="searchbtn"
                            onClick={() => onSearch(search.trim())}>Search</button>
                    </form>
                </div>
            </div>
        </nav>
    )
}

export default Head