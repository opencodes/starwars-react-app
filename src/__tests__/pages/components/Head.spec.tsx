import Head from '../../../pages/components/Head';
import { renderWithRouter } from "../../test-utils";
import { fireEvent, screen } from '@testing-library/react';

describe('Head', () => {
    test('should render Head component correctly', () => {
        const props = {
            onSearch: jest.fn(),
        }
        const { asFragment } = renderWithRouter(<Head {...props} />);
        expect(asFragment).toMatchSnapshot();
    });
    test('should match input value on change event', () => {
        const props = {
            onSearch: jest.fn(),
        }
        renderWithRouter(<Head {...props} />);
        const input = screen.getByTestId('searchinput')
        const searchbtn = screen.getByTestId('searchbtn')
        fireEvent.change(input, { target: { value: 'Luke' } })
        fireEvent.click(searchbtn)
        expect(input.value).toEqual("Luke");
    });
});

