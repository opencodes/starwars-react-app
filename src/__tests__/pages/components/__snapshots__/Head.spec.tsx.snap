// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Head should render Head component correctly 1`] = `[Function]`;
